nav2_smoother/src/savitzky_golay_smoother.cpp

```diff
@@ -71,11 +77,16 @@ bool SavitzkyGolaySmoother::smooth(
         RCLCPP_WARN(
           logger_,
           "Smoothing time exceeded allowed duration of %0.2f.", max_time.seconds());
-        return false;
+        throw nav2_core::SmootherTimedOut("Smoothing time exceed allowed duration");
+      }
+
+      bool segment_was_smoothed = smoothImpl(curr_path_segment, reversing_segment);
```

Comment:
smoothImpl always returns true, these changes are a bit redundant

Reply:
But are useful in the Simple smoother, I think just here in the SG-Smoother its unnecessary.

Reply:
Oh?

Why even return a bool if it is always true?

Reply:
The SG-Smoother was a 2 day project and I started with the simple Smoother as a base, so just an overlooked thing